src/aproxy.py

```diff
@@ -0,0 +1,294 @@
+                continue
+            try:
+                # Check if entry is a valid IP or CIDR
+                ipaddress.ip_network(entry, strict=False)
+                ip_list.append(entry)
+            except ValueError:
+                # Not an IP, attempt to resolve as hostname
+                resolved_ips = self._resolve_hostname_to_ips(entry)
+                ip_list.extend(resolved_ips)
+        return ip_list
+
+    def _render_nftables_rules(self) -> str:
+        """Render nftables rules for transparent proxy interception.
+
+        - Redirect outbound traffic on configured intercept_ports to aproxy.
+        - Exclude private and loopback ranges.
+        - Drop inbound traffic to aproxy listener to prevent reflection attacks.
+
+        Returns:
+            The nftables rules as a string.
+        """
+        server_ip = self._get_primary_ip()
+        ports_clause = ", ".join(self.config.intercept_ports_list)
+        excluded_ips = [
+            "127.0.0.0/8",
+            "10.0.0.0/8",
+            "172.16.0.0/12",
+            "192.168.0.0/16",
+        ] + self._get_no_proxy_ips()
+
+        return f"""#!/usr/sbin/nft -f
+        table ip aproxy
+        flush table ip aproxy
+        table ip aproxy {{
+            set excluded_nets {{
+                    type ipv4_addr;
+                    flags interval; auto-merge;
+                    elements = {{ {', '.join(excluded_ips)} }}
+                }}
+            chain prerouting {{
+                type nat hook prerouting priority dstnat; policy accept;
+                ip daddr @excluded_nets return
+                tcp dport {{ {ports_clause} }} counter dnat {server_ip}:{APROXY_LISTEN_PORT}
+            }}
+
+            chain output {{
+                type nat hook output priority -150; policy accept;
+                ip daddr @excluded_nets return
+                tcp dport {{ {ports_clause} }} counter dnat to {server_ip}:{APROXY_LISTEN_PORT}
+            }}
+
+            chain input {{
+                type filter hook input priority filter; policy accept;
+                tcp dport {APROXY_LISTEN_PORT} drop
+            }}
+        }}
+        """
+
+    def write_nft_config(self) -> None:
+        """Write nft config to disk."""
+        NFT_CONF_FILE.parent.mkdir(parents=True, exist_ok=True)
+        NFT_CONF_FILE.write_text(textwrap.dedent(self._render_nftables_rules()), encoding="utf-8")
+
+    def apply_nft_config(self) -> None:
+        """Apply nft config immediately.
+
+        Raises:
+            CalledProcessError: If the nft command fails.
+        """
+        try:
+            # nosec B404,B603,B607: calling trusted system binary with predefined args
+            subprocess.run(["nft", "-f", str(NFT_CONF_FILE)], check=True)  # nosec
+            logger.info("Applied nftables rules successfully.")
+        except subprocess.CalledProcessError as e:
+            logger.error("Failed to apply nftables rules: %s", e)
+            raise
```

Comment:
Create a business exception with an explicit error message (specifying the path to the file).
The logging is unnecessary because the uncaught exception will anyway log the error